Shared/Sources/CollectableCollection.swift

```diff
@@ -22,35 +28,27 @@ struct CollectableCollection<CollectableCollectionObject: Collectable>: Collecti
     return self[indexPath.section][indexPath.row]
   }
 
-  typealias FilterClosure = ((CollectableCollectionObject) -> Bool)?
-  func filteredCollectableCollection(itemFilter: FilterClosure = nil) -> CollectableCollection {
-    if let filter = itemFilter {
-      let groups = self.items.map({ group -> CollectableGroup<CollectableCollectionObject> in
-        // Runs through every `CollectableGroup` in order to filter items using the received closure.
-        let items = group.items.filter({ object -> Bool in
-          filter(object)
-        })
-        return CollectableGroup<CollectableCollectionObject>(title: group.title, items: items)
-        // After filtering items in each group, we need to filter out empty `CollectableGroup`s.
-        }).filter({ group -> Bool in
-          return group.items.count > 0
-        })
-      // Returns a new `CollectableCollection` with only matching items and groups containing them.
-      return CollectableCollection(groups: groups)
-    } else {
-      return self
-    }
+  typealias FilterClosure = ((CollectableCollectionObject) -> Bool)
+  func filteredCollectableCollection(itemFilter: FilterClosure) -> CollectableCollection {
+    let groups = self.items.map({ group -> CollectableGroup<CollectableCollectionObject> in
+      // Runs through every `CollectableGroup` in order to filter items using the received closure.
+      let items = group.items.filter({ object -> Bool in
```

Comment:
Can't you just pass `itemFilter` to the `filter` method?  No need to have a closure call a closure with the same signature.